provider/datagen/src/tests/make_testdata.rs

```diff
@@ -55,11 +65,63 @@ fn generate_json_and_verify_postcard() {
         ])
         .export(
             &DatagenProvider::new_testing(),
-            MultiExporter::new(vec![json_out, postcard_out]),
+            MultiExporter::new(vec![json_out, postcard_out, stubdata_out]),
         )
         .unwrap();
 }
 
+/// Generates a stub data directory that can be used with `ICU4X_DATA_DIR`
+/// for faster development and debugging. For example, put the following in
+/// VSCode settings.json:
+///
+/// ```javascript
+/// "rust-analyzer.cargo.extraEnv": {
+///   // Relative to provider/baked/x/src
+///   "ICU4X_DATA_DIR": "../../../datagen/tests/data/stub"
+/// },
+/// ```
+struct BakedStubdataExporter(baked_exporter::BakedExporter);
+
+impl DataExporter for BakedStubdataExporter {
+    fn put_payload(
+        &self,
+        _key: DataKey,
+        _locale: &DataLocale,
+        _payload: &DataPayload<ExportMarker>,
+    ) -> Result<(), DataError> {
+        // do not put any payloads in stubdata!
+        Ok(())
+    }
+
+    fn flush_singleton(
+        &self,
+        key: DataKey,
+        payload: &DataPayload<ExportMarker>,
+    ) -> Result<(), DataError> {
+        self.0.flush_singleton(key, payload)
```

Comment:
Note that the singleton keys are still being exported. If I don't export them or if I export them as non-singleton, I get errors like

```
no associated item named `SINGLETON_LOCID_TRANSFORM_ALIASES_V1` found for struct `Baked` in the current scope
associated item not found in `Baked`
```

I'm not sure how to work around this other than to require that all data structs implement `Default` and then plumb through a function that resets the payload to `Default`.

Reply:
This is kind of the point of singleton keys, the data structs has to be present. 

I think implementing `Default` would be very bad, as that would be part of the public API. We don't want anyone to construct a data struct like that. You'd also have to pipe this through the `dyn Exportable`, which means either all structs have to implement `Default`, or you need custom filtering code in `icu_provider`. I think this is a no-go.

You can implement custom filtering here by matching on keys:
```rust
self.0.flush_singleton(
  key,
  match key {
    LineBreakDataV1Marker:KEY => &DataPayload::from_owned(...).upcast(),
    // ...
    _ => payload
  } 
}
```

Reply:
Now that we're including `und` for all keys it is no longer inconsistent to include it for singleton keys so I consider this thread to be obsolete.

Reply:
How is the build time? We do have some fairly big `und`s and singletons

Reply:
The build time is significantly smaller than it had been before so I'm happy. The slowest crate to build before was icu_datetime, which has a lot of small data, and icu_datetime is a lot faster with this change.